bitcoin/src/witness_program.rs

```diff
@@ -55,64 +80,285 @@ impl WitnessProgram {
+            }
+        }
+
+        const INVALID_LENGTH_PROGRAM_TOO_LONG: &[u8] = &[0x00, 0x29,
+            0x75, 0x1e, 0x76, 0xe8, 0x19,
+            0x91, 0x96, 0xd4, 0x54, 0x94,
+            0x1c, 0x45, 0xd1, 0xb3, 0xa3,
+            0x23, 0xf1, 0x43, 0x3b, 0xd6,
+            0x75, 0x1e, 0x76, 0xe8, 0x19,
+            0x91, 0x96, 0xd4, 0x54, 0x94,
+            0x1c, 0x45, 0xd1, 0xb3, 0xa3,
+            0x23, 0xf1, 0x43, 0x3b, 0xd6,
+            0x00];
+
+        #[test]
+        fn new_invalid_program_length_too_long() {
+            match WitnessProgram::new(INVALID_LENGTH_PROGRAM_TOO_LONG) {
+                Ok(_) => assert_eq!(None, Some("Error, program created with invalid version")),
+                Err(e) => assert_eq!(WitnessProgramError::InvalidProgramLength(41), e)
+            }
+        }
+    }
+
+    mod test_to_scriptpubkey {
+        use crate::witness_program::WitnessProgram;
+
+        #[test]
+        fn to_scriptpubkey_valid_p2sh_p2wpkh_program() {
+            let witness_program = WitnessProgram {
+                version: 0x00,
+                program: vec![
+                    0x75, 0x1e, 0x76, 0xe8, 0x19,
+                    0x91, 0x96, 0xd4, 0x54, 0x94,
+                    0x1c, 0x45, 0xd1, 0xb3, 0xa3,
+                    0x23, 0xf1, 0x43, 0x3b, 0xd6
+                ],
+            };
+            assert_eq!(vec![0x00, 0x14,
+                            0x75, 0x1e, 0x76, 0xe8, 0x19,
+                            0x91, 0x96, 0xd4, 0x54, 0x94,
+                            0x1c, 0x45, 0xd1, 0xb3, 0xa3,
+                            0x23, 0xf1, 0x43, 0x3b, 0xd6
+            ], witness_program.to_scriptpubkey());
+        }
+
+        #[test]
+        fn to_scriptpubkey_valid_p2sh_p2wsh_program() {
+            let witness_program = WitnessProgram {
+                version: 0x00,
+                program: vec![0x18, 0x63, 0x14, 0x3c, 0x14,
+                              0xc5, 0x16, 0x68, 0x04, 0xbd,
+                              0x19, 0x20, 0x33, 0x56, 0xda,
+                              0x13, 0x6c, 0x98, 0x56, 0x78,
+                              0xcd, 0x4d, 0x27, 0xa1, 0xb8,
+                              0xc6, 0x32, 0x96, 0x04, 0x90,
+                              0x32, 0x62
+                ],
+            };
+            assert_eq!(vec![0x00, 0x20,
+                            0x18, 0x63, 0x14, 0x3c, 0x14,
+                            0xc5, 0x16, 0x68, 0x04, 0xbd,
+                            0x19, 0x20, 0x33, 0x56, 0xda,
+                            0x13, 0x6c, 0x98, 0x56, 0x78,
+                            0xcd, 0x4d, 0x27, 0xa1, 0xb8,
+                            0xc6, 0x32, 0x96, 0x04, 0x90,
+                            0x32, 0x62
+            ], witness_program.to_scriptpubkey());
+        }
+
+        #[test]
+        fn to_scriptpubkey_valid_op_1_program() {
+            let witness_program = WitnessProgram {
+                version: 0x01,
+                program: vec![0x75, 0x1e, 0x76, 0xe8, 0x19,
```

Comment:
Adhere to the test suite convention of providing a slice containing all programs at the top of the inner mod.

See https://github.com/ArgusHQ/wagyu/blob/v0.6.0/bitcoin/src/extended_private_key.rs#L682